magma/circuit.py

```diff
@@ -665,13 +665,13 @@ def __new__(metacls, name, bases, dct):
 
     def check_unconnected(self):
         for port in self.interface.inputs():
-            if not port.driven():
```

Comment:
why should this be trace vs driven?

Reply:
With the change to the driven() interface, a port can be driven by a temporary (driven() is True), but tracing it to a source returns None (it's not driven by an output)